### PR TITLE
Add warnings to MD5 and SHA1 hash documentation

### PR DESCRIPTION
**Pull request by RX14**
Fixes https://github.com/crystal-lang/crystal/issues/2743.

I'm not 100% on the wording of the warnings.

### CODE REVIEW

File: src/digest/md5.cr [hunk 1]
```diff
@@ -1,6 +1,9 @@
 require "./base"
 
 # Implements the MD5 digest algorithm.
+#
+# Warning: MD5 is no longer a cryptograpically safe hash, and should not be
```
Comment (ysbaddaden) [inline src/digest/md5.cr]: typeo: "cryptographically"

Comment (RX14) [inline src/digest/md5.cr]: Done

**Comment by Papierkorb** [comment]
The term should be `cryptographically secure`, not "safe".
The `secure applications` isn't clear enough to me, people will never assume that what they're doing can have security implications.  What about something along the line of `.. and should not be used in security-related components, like password checks`.  Still not awesome, but hopefully makes it more clear that one of the primary use-cases, password hashing, isn't a good fit.

**Comment by mverzilli** [comment]
Thanks @RX14!

**Pull request merged**